### PR TITLE
Add json validator

### PR DESCRIPTION
**Pull request by exileed**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | no

### CODE REVIEW

File: tests/framework/validators/JsonValidatorTest.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        $string = '{ "json": "string" }';
+        $val = new JsonValidator();
+        $this->assertTrue($val->validate($string));
+        $this->assertFalse($val->validate('5'));
+        $this->assertFalse($val->validate(null));
+        $this->assertFalse($val->validate([]));
+    }
+
+    public function testErrorMessage()
```
Comment (githubjeka) [inline tests/framework/validators/JsonValidatorTest.php]: How about
->validate('')
->validate(' ')
->validate(false)
->validate(null)

Comment (exileed) [reply tests/framework/validators/JsonValidatorTest.php]: Add in test

File: src/validators/JsonValidator.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function validateValue($value): ?array
+    {
+        if ( ! is_scalar($value) && ! method_exists($value, '__toString')) {
+            return null;
```
Comment (cebe) [inline src/validators/JsonValidator.php]: should be considered invalid in this case, i.e. return error message, not null.

**Comment by samdark** [comment]
Does that make sense to have a validator that validates just syntax? How about https://github.com/justinrainbow/json-schema or https://github.com/swaggest/php-json-schema?

**Comment by cebe** [comment]
validating JSON-Schema is a lot more useful than validating just the fact that it is json.

**Comment by samdark** [comment]
It is not a simple thing though and I'm not sure we want either to implement it ourselves or having core that depends on any of the libraries that do.

**Comment by lubosdz** [comment]
Then other guys will also want XML/XSD/WSDL validators ...

**Comment by samdark** [comment]
Yeah. I think we should not implement it.

**Pull request closed without merging**